### DISCUSSION TITLE
EagleTree Digital I2C Airspeed Sensor

### DISCUSSION THREAD

**Discussion by james-au** [iNavFlight/inav#10125] [General]
There is an I2C based digital airspeed pitot sensor that was available with EagleTree.  I used this airspeed sensor for years during my analog FPV days and the airspeed readings are accurate.

Can this EagleTree digital airspeed sensor be used with iNav?  So far I only got partially there.

The EagleTree airspeed sensor is 4-pin with 5V, SDA, SDC, GND pins.  Here a 2-page overview of how to interface with this sensor and includes sample C++ function to read values from this airspeed sensor:

https://www.manualslib.com/manual/2589541/Eagle-Tree-Systems-Eagle-Tree-Microsensor-V3.html?page=2#manual

I have hooked this airspeed sensor up to my SpeedyBee F405 Wing and the sensor powers on correctly.  It is using I2C device ID 0xEA which may be different than the MS4525 digital speed sensor which is the only I2C digital speed sensor available in the drop down list in the "Configuration" tab.

iNav never detects the EagleTree airspeed sensor and I've used pull-up resistors on the SDA and SDC lines as recommended too.  Possibly iNav simply is not aware of the EagleTree slave I2C ID.  So iNav also boots with constant beeping and shows "hardware error" so it detects something is there but can't identify it.

Could be as simple as adding an "EagleTree" option in the Pitot pull down?  It would be identical to currently available MS4525 option but updated with EagleTree 0xEA slave I2C ID.

Would be great to have iNav work with this EagleTree digital speed sensor because these sensors are not cheap and I have several of them available.  Many old school FPV pilots may have a lot of old EagleTree sensors as well.

<img width="581" alt="image" src="https://github.com/iNavFlight/inav/assets/15576965/c36798fb-740b-4b45-b9a0-14fec46fc502">

**Comment by JetsRfaster** [comment]
+1 - I have one of these and would like to use it also 👍
